File: appli-android/app/src/main/java/com/example/sportapp/core/data/model/PlannedWorkoutExercise.kt
package com.example.sportapp.core.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.ColumnInfo
import androidx.room.ForeignKey
import com.example.sportapp.core.utils.CustomDateUtils.getNowISO8601
import java.util.UUID

@Entity(
    tableName = "planned_workout_exercises",
    indices = [androidx.room.Index(value = ["uuid"], unique = true)],
    foreignKeys = [
        ForeignKey(
            entity = PlannedWorkout::class,
            parentColumns = ["uuid"],
            childColumns = ["planned_workout_uuid"],
            onDelete = ForeignKey.CASCADE
        ),
        ForeignKey(
            entity = Exercise::class,
            parentColumns = ["uuid"],
            childColumns = ["exercise_uuid"],
            onDelete = ForeignKey.CASCADE
        )
    ]
)
data class PlannedWorkoutExercise(
    @PrimaryKey
    @ColumnInfo(name = "uuid") val uuid: String,

    @ColumnInfo(name = "planned_workout_uuid") val plannedWorkoutUUID: String,
    @ColumnInfo(name = "exercise_uuid") val exerciseUUID: String,
    val sets: Int,
    val reps: String,
    val phase: String,   // "WARMUP", "TRAINING", "POST_TRAINING"
    val status: String,  // "PLANNED", "DONE", "NOT_STARTED", "SKIPPED"
    @ColumnInfo(name = "order") val order: Int = 0,
    val ignored: Boolean = false,

    @ColumnInfo(defaultValue = "0") val synced: Boolean = false,
    @ColumnInfo(defaultValue = "0") val pendingDeletion: Boolean = false,

    @ColumnInfo(name = "updated_at") val updatedAt: String? = getNowISO8601(),
)

File: serveur/app/models/planned_workout_exercise.py
# Exemple : app/models/planned_workout_exercise.py

from sqlalchemy import Column, Integer, ForeignKey, String, Boolean, DateTime, func
from app.database import Base

import uuid

class PlannedWorkoutExercise(Base):
    __tablename__ = "planned_workout_exercises"

    id = Column(Integer, primary_key=True, index=True)
    planned_workout_uuid = Column(String, ForeignKey("planned_workouts.uuid", ondelete="CASCADE"), nullable=False, index=True)
    exercise_uuid = Column(String, ForeignKey("exercises.uuid", ondelete="CASCADE"), nullable=False, index=True)

    sets = Column(Integer, nullable=False)
    reps = Column(String, nullable=False)
    phase = Column(String, nullable=False, default="TRAINING")
    status = Column(String, nullable=False, default="PLANNED")
    order = Column(Integer, nullable=False, default=0)
    ignored = Column(Boolean, default=False, nullable=False)

    uuid = Column(String, default=lambda: str(uuid.uuid4()), nullable=False, unique=True)

    updated_at = Column(DateTime(timezone=True), nullable=True)